src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -646,16 +662,30 @@ public void doStatus(@QueryParameter String token) {
             System.err.println("Checking status");
             checkToken(token);
         }
+        /**
+         * Used to run test methods on a separate thread so that code that uses {@link Stapler.getCurrentRequest}
+         * does not inadvertently interact with the request for {@link #doStep} itself.
+         */
+        private static final ExecutorService STEP_RUNNER = Executors.newSingleThreadExecutor(
+                new NamingThreadFactory(new DaemonThreadFactory(), RealJenkinsRule.class.getName() + ".STEP_RUNNER"));
```

Comment:
Not sure whether the thread factories matter here. Should we use `ClassLoaderSanityThreadFactory`? Do we even care if this is a daemon thread or not?

Reply:
> Do we even care if this is a daemon thread or not?

Likely not.

Reply:
Switched to a regular thread pool in [869ed42](https://github.com/jenkinsci/jenkins-test-harness/pull/422/commits/869ed42113bfc9b6f3d3a4f70e26fa1841e7fc3e).